aws/resource_aws_media_convert_queue.go

```diff
@@ -0,0 +1,264 @@
+
+	createOpts := &mediaconvert.CreateQueueInput{
+		Name:        aws.String(d.Get("name").(string)),
+		Status:      aws.String(d.Get("status").(string)),
+		PricingPlan: aws.String(d.Get("pricing_plan").(string)),
+		Tags:        tagsFromMapGeneric(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		createOpts.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("reservation_plan_settings"); ok {
+		reservationPlanSettings := v.([]interface{})[0].(map[string]interface{})
+		createOpts.ReservationPlanSettings = expandMediaConvertReservationPlanSettings(reservationPlanSettings)
+	}
+
+	resp, err := conn.CreateQueue(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating Media Convert Queue: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.Queue.Name))
+
+	return resourceAwsMediaConvertQueueRead(d, meta)
+}
+
+func resourceAwsMediaConvertQueueRead(d *schema.ResourceData, meta interface{}) error {
+	originalConn := meta.(*AWSClient).mediaconvertconn
+
+	endpointURL, err := getAwsMediaConvertEndpoint(originalConn)
+	if err != nil {
+		return fmt.Errorf("Error getting Media Convert Endpoint: %s", err)
+	}
+
+	sess, err := session.NewSession(&originalConn.Config)
+	if err != nil {
+		return fmt.Errorf("Error creating AWS session: %s", err)
+	}
+	conn := mediaconvert.New(sess.Copy(&aws.Config{Endpoint: aws.String(endpointURL)}))
+
+	getOpts := &mediaconvert.GetQueueInput{
+		Name: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetQueue(getOpts)
+	if isAWSErr(err, mediaconvert.ErrCodeNotFoundException, "") {
+		log.Printf("[WARN] Media Convert Queue (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error getting Media Convert Queue: %s", err)
+	}
+
+	d.Set("arn", resp.Queue.Arn)
+	d.Set("name", resp.Queue.Name)
+	d.Set("description", resp.Queue.Description)
+	d.Set("pricing_plan", resp.Queue.PricingPlan)
+	d.Set("status", resp.Queue.Status)
+
+	if err := d.Set("reservation_plan_settings", flattenMediaConvertReservationPlan(resp.Queue.ReservationPlan)); err != nil {
+		return fmt.Errorf("Error setting Media Convert Queue reservation_plan_settings: %s", err)
+	}
+
+	if err := saveTagsMediaConvert(conn, d, aws.StringValue(resp.Queue.Arn)); err != nil {
+		return fmt.Errorf("Error setting Media Convert Queue tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsMediaConvertQueueUpdate(d *schema.ResourceData, meta interface{}) error {
+	originalConn := meta.(*AWSClient).mediaconvertconn
+
+	endpointURL, err := getAwsMediaConvertEndpoint(originalConn)
+	if err != nil {
+		return fmt.Errorf("Error getting Media Convert Endpoint: %s", err)
+	}
+
+	sess, err := session.NewSession(&originalConn.Config)
+	if err != nil {
+		return fmt.Errorf("Error creating AWS session: %s", err)
+	}
+	conn := mediaconvert.New(sess.Copy(&aws.Config{Endpoint: aws.String(endpointURL)}))
+
+	updateOpts := &mediaconvert.UpdateQueueInput{
```

Comment:
This section of code should only execute when one of the relevant attributes changes and not when there are only tags updates, e.g.

```go
if d.HasChange("description") || d.HasChange("reservation_plan_settings") || d.HasChange("status") {
	updateOpts := &mediaconvert.UpdateQueueInput{
//...
	if err != nil {
		return fmt.Errorf("error updating MediaConvert Queue (%s): %s", d.Id(), err)
	}
}

if d.HasChange("tags") {
// ...
```